### DISCUSSION TITLE
Is there any way of extracting text as blocks?

### DISCUSSION THREAD

**Discussion by dmlls** [jsvine/pdfplumber#673] [Q&A]
Hi everyone 👋🏻

I have been going through the documentation, issues and discussions but I haven't found an answer to the following question:

**Does pdfplumber provide any way of extracting text as blocks?** Something along the lines of [`Page.get_text("blocks")`](https://pymupdf.readthedocs.io/en/latest/page.html?highlight=get_text#Page.get_text) in PyMuPDF.

(Sorry in advance if this question has already been answered elsewhere.)

**Comment by jsvine** [comment]
Hi @dmlls, and thanks for your interest in this library. I'm not very familiar with MyMuPDF's get-blocks method, but adding something like that to `pdfplumber` is intriguing. (There isn't currently an equivalent method.) For single-column text, you could do something like this:

```python
blocks = re.split(r"\n\n+", my_page.extract_text(layout=True, ...))
```

... but multi-column layouts would require a more sophisticated approach.
